test/integration/evictions/evictions_test.go

```diff
@@ -334,6 +340,85 @@ func TestEvictionVersions(t *testing.T) {
+	cases := map[string]struct {
+		enablePodDisruptionConditions bool
+		phase                         v1.PodPhase
+	}{
+		"terminal pod with job backoff policy enabled": {
+			enablePodDisruptionConditions: true,
+			phase:                         v1.PodSucceeded,
+		},
+		"terminal pod with job backoff policy disabled": {
+			enablePodDisruptionConditions: false,
+			phase:                         v1.PodSucceeded,
+		},
+		"running pod with job backoff policy enabled": {
+			enablePodDisruptionConditions: true,
+			phase:                         v1.PodRunning,
+		},
+		"running pod with job backoff policy disabled": {
+			enablePodDisruptionConditions: false,
+			phase:                         v1.PodRunning,
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			closeFn, rm, informers, _, clientSet := rmSetup(t)
+			defer closeFn()
+
+			ns := framework.CreateNamespaceOrDie(clientSet, "eviction-with-finalizers", t)
+			defer framework.DeleteNamespaceOrDie(clientSet, ns, t)
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.PodDisruptionConditions, tc.enablePodDisruptionConditions)()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			informers.Start(ctx.Done())
+			go rm.Run(ctx)
+
+			pod := newPod("pod")
+			pod.ObjectMeta.Finalizers = []string{"test.k8s.io/finalizer"}
+			if _, err := clientSet.CoreV1().Pods(ns.Name).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+				t.Errorf("Failed to create pod: %v", err)
+			}
+
+			pod.Status.Phase = tc.phase
+			addPodConditionReady(pod)
+			if _, err := clientSet.CoreV1().Pods(ns.Name).UpdateStatus(ctx, pod, metav1.UpdateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			waitToObservePods(t, informers.Core().V1().Pods().Informer(), 1, tc.phase)
+			deleteOption := metav1.DeleteOptions{}
+
+			eviction := newV1Eviction(ns.Name, pod.Name, deleteOption)
+
+			err := wait.PollImmediate(5*time.Second, 60*time.Second, func() (bool, error) {
+				e := clientSet.PolicyV1().Evictions(ns.Name).Evict(ctx, eviction)
+				if e != nil {
+					return false, e
+				}
+				return true, nil
+			})
+			if err != nil {
+				t.Fatalf("Eviction of pod failed %v", err)
+			}
+
+			updatedPod, e := clientSet.CoreV1().Pods(ns.Name).Get(ctx, pod.Name, metav1.GetOptions{})
+			if e != nil {
+				t.Fatalf("Failed to get the pod %q with error: %q", klog.KObj(pod), e)
+			}
+			_, cond := podutil.GetPodCondition(&updatedPod.Status, v1.PodConditionType(v1.AlphaNoCompatGuaranteeDisruptionTarget))
+			if tc.enablePodDisruptionConditions == true && cond == nil {
+				t.Errorf("Pod %q does not have the expected condition: %v", klog.KObj(updatedPod), v1.AlphaNoCompatGuaranteeDisruptionTarget)
+			} else if tc.enablePodDisruptionConditions == false && cond != nil {
+				t.Errorf("Pod %q does has the unexpected condition: %v", klog.KObj(updatedPod), v1.AlphaNoCompatGuaranteeDisruptionTarget)
```

Comment:
nit: `Pod %q has an unexpected condition: %v`

Reply:
Thanks, updated in a couple of places.